aya/src/obj/mod.rs

```diff
@@ -510,13 +509,24 @@ fn parse_map(section: &Section, name: &str) -> Result<Map, ParseError> {
 }
 
 fn parse_map_def(name: &str, data: &[u8]) -> Result<bpf_map_def, ParseError> {
-    if mem::size_of::<bpf_map_def>() > data.len() {
+    if data.len() > mem::size_of::<bpf_map_def>() {
         return Err(ParseError::InvalidMapDefinition {
             name: name.to_owned(),
         });
     }
 
-    Ok(unsafe { ptr::read_unaligned(data.as_ptr() as *const bpf_map_def) })
+    if data.len() <  mem::size_of::<bpf_map_def>() {
+        let mut map_def = bpf_map_def { ..Default::default() };
+        unsafe {
+            ptr::copy(data.as_ptr() as *const bpf_map_def, &mut map_def as *mut bpf_map_def, 1);
+            // id and pinning will be garbage data
```

Comment:
This is UB, see https://doc.rust-lang.org/stable/std/ptr/fn.copy.html#safety

You need to cast both to bytes and copy bytes. After copying bytes, resetting id and pinning won't be necessary.

Reply:
I was trying to avoid that byte copy loop but you're right - no guarantees about what will happen when copying past the end of `data`.

See 1404d12a7e29ab02225bc13fd36794f6c6206db1